### ISSUE TITLE
Unable to unlock dislocker volume at boot

### ISSUE THREAD

**Issue by xi-pinmping** [roddhjav/apparmor.d#137]
Recently after installing and setting up apparmor and these profiles, I have been unable to decrypt and mount my dislocker volume at boot and even running the commands in the command line keeps it from mounting properly. I am not certain this is an apparmor issue but after seeing the known issue with udev, I believe it could be the issue. 

Im posting more to get a second opinion on my assumption than a fix for my issue since I know the behavior of sytemd is out of scope here. I didn't think it would be related however, as I didn't enable any profiles at that level, only stuff like mpv, libreoffice, and brave. 

Additionally, this is an unrelated issue but is checking lsb-release really necessary? I originally installed archlabs but only use the themeing at this point, and have merged cachyos into my install but keep archlabs for the neofetch logo lol. Its dumb but also quite annoying to have to edit lsb-release to install and update. 

Thanks for your time and service to the communtiy!

**Comment by roddhjav** [comment]
Can you have a look at your apparmor log (`aa-log`, `aa-log -f 1`...)? If your issue is apparmor related it should be seen in the log. Another good test that may be useful for udev: try disable apparmor (and reboot), if you don't have the issue again it was apparmor.

I am confused. You initially said you only installed some profile, and after you mentioned the package install, which install everything.

> Additionally, this is an unrelated issue but is checking lsb-release really necessary?

I understand your point and even agree to it. The configure script needs to know the distribution (or for arch, the base distribution) in order to apply distribution specific configuration. If you have another easy way to get this info, I am open to idea.

**Comment by xi-pinmping** [comment]
I ended up turing off bitlocker to see if having it plain text would help but the issue ended up persisting almost identically. The (former) bitlocker partition is on a microSD card and has my media so I mount it at boot if its there and link it to my music/videos directories. After fighting with it until 1am I finally gave up and disabled apparmor/auditd and restarted and everything mounted fine, although it is now in plain text. 

To elaborate your confusion, this week I started setting up apparmor. I got it installed with base profiles and all was working well. But the next day after I installed apparmor.d-git from the AUR, that is when my issues started occurring. It wasn't the first off on cycle but the second. I thought initially I messed something up in my fstab but after investigating initially, nothing had really been changed and even removing the non-pertinent options I added it still didn't fix anything. Sorry for the confusion, it has been a long and confusing road lol.  

[aa-log.txt](https://github.com/roddhjav/apparmor.d/files/11122350/aa-log.txt)

**Comment by xi-pinmping** [comment]
I have just now confirmed it is an issue with apparmor somehow. After disabling encryption entirely and still having issues mounting at boot, I reencrypted the partition and disabled apparmor/auditd and can mount at boot no issues again. Still quite puzzled though, as I didn't set any profiles having to do with boot or systemd to enforce yet. 

I will continue testing in my spare time to determine if this is a stock apparmor issue or an issue with this package, but at the moment I am quite tied up.

**Comment by roddhjav** [comment]
I have done some minor edit from your logs concerning `udisksd`, you may give it a try.

Usually, a profile in complain mode cannot break the program it confines.
However, there are some **major exceptions**:

* `deny` rules are enforced even in complain mode,
* `attach_disconnected` (and `the mediate_deleted`) will break the program if 
  they are required and missing in the profile,
* If apparmor does not find the profile to transition `rPx`.

**Comment by xi-pinmping** [comment]
I just updated and tried re-enabling but sadly no change. I did find another udisks entry. Admittedly, I am not the most knowledgeable with apparmor but I'm working on it lol. But I noticed another udisks entry in logs after checking again. Will attach them below. But will two udisk entries be governed by the same profile if they are named differently? I know they can have dependencies, but when I was working on getting profiles to play nice with brave I noticed I wasn't able to get it to set '/usr/bin/brave//null-/usr/bin/bash//null-/opt/brave-bin/brave' to enforce. The separate udisk entry in log is similar to this as it is called 'udisksd//null-/usr/bin/mount.exfat-fuse'. Could this be the offender? The partition it is failing to mount is an exfat partition. 

[aa-log2.txt](https://github.com/roddhjav/apparmor.d/files/11125029/aa-log2.txt)
